ckanext/switzerland/harvester/base_sbb_harvester.py

```diff
@@ -825,89 +823,56 @@ def _import_stage(self, harvest_object):  # noqa
-
-                # always overwrite this metadata
-                file_format, mimetype, mimetype_inner = self._get_mimetypes(f)
-                resource_meta["format"] = file_format
-                resource_meta["mimetype"] = mimetype
-                resource_meta["mimetype_inner"] = mimetype_inner
-
-                resource_meta["name"] = {
-                    "de": os.path.basename(f),
-                    "en": os.path.basename(f),
-                    "fr": os.path.basename(f),
-                    "it": os.path.basename(f),
-                }
-                resource_meta["title"] = {
-                    "de": os.path.basename(f),
-                    "en": os.path.basename(f),
-                    "fr": os.path.basename(f),
-                    "it": os.path.basename(f),
-                }
-
-                resource_meta["issued"] = now
-                resource_meta["version"] = now
+            resource_meta = self.resource_dict_meta
 
-                # take this metadata from the old version if available
-                resource_meta["rights"] = resource_meta.get("rights", "")
-                resource_meta["license"] = resource_meta.get("license", "")
-                resource_meta["coverage"] = resource_meta.get("coverage", "TODO")
-                resource_meta["description"] = resource_meta.get(
-                    "description", {"de": "", "en": "", "fr": "", "it": ""}
-                )
-                resource_meta["relations"] = resource_meta.get("relations", [])
+            resource_meta["identifier"] = os.path.basename(f)
 
-                log_msg = "Creating new resource: %s"
+            file_format, mimetype, mimetype_inner = self._get_mimetypes(f)
+            resource_meta["format"] = file_format
+            resource_meta["mimetype"] = mimetype
+            resource_meta["mimetype_inner"] = mimetype_inner
 
-            # -----------------------------------------------------
-            # create the resource, but use the known metadata (of the old resource)
-            # -----------------------------------------------------
-            else:
-                old_resource_id = resource_meta["id"]
+            resource_meta["name"] = {
+                "de": os.path.basename(f),
+                "en": os.path.basename(f),
+                "fr": os.path.basename(f),
+                "it": os.path.basename(f),
+            }
+            resource_meta["title"] = {
+                "de": os.path.basename(f),
+                "en": os.path.basename(f),
+                "fr": os.path.basename(f),
+                "it": os.path.basename(f),
+            }
 
-                self._reset_resource(resource_meta)
+            resource_meta["issued"] = now
+            resource_meta["version"] = now
 
-                log_msg = "Updating resource (with known metadata): %s"
+            # take this metadata from the old version if available
+            resource_meta["rights"] = old_resource_meta.get("rights", "")
```

Comment:
I am wondering whether we should only set those fields from the old_resource_meta, as fallback. sounds cumbersome, to check twice, whether that data is in the old metadata, but I think it makes the most sense. WDYT?

Reply:
Not sure what you mean here, sorry.

Reply:
what I mean is, something along those lines:
resource_meta["rights"] = resource_meta.get("rights", old_resource_meta['rights'])

but not sure whether that makes sense

Reply:
oh right, I see!

(thinking out loud) the 'blank slate' resource metadata in the harvester looks like this:

```
    resource_dict_meta = {
        "state": "active",
        "rights": "",
        "license": "",
        "coverage": "Coverage",
    }
```
and here we are trying to set these fields from the old metadata: rights, license, coverage, description, relations, rights... huh, the old code was setting `rights` twice and so is this new code. Oops, I'll fix that.

Anyway, we already have `""` for `rights` in `resource_meta`, so I think it doesn't make sense to specifically get `resource_meta["rights"]` and fall back to `old_resource_meta["rights"]`, because there's no point getting just an empty string. `license` is an empty string too.

`coverage` is a placeholder value, which it seems like we replace with a different placeholder value (`"TODO"`), which is just ugly tbh. It looks like we never get a real value for coverage at all, and we don't display it on the resource or dataset pages either. I guess it's there because it's in DCAT-AP CH v1?

Thank you for bearing with my rambling. :sweat_smile: Summary:

- I don't think it's worth getting these values from the new `resource_meta`, but the way we're doing it now is quite clunky and I want to tidy it up a bit
- Also, I want to remove the double setting of `rights`

I'll do those things tomorrow if you agree, it's way too late today.

Reply:
thank you for taking the time to write down your thoughts. I think that makes sense to me, thank you for explaining this again to me 😄 
Yes, I think tomorrow makes more sense 😉